controller/adoptopenebs/reconciler.go

```diff
@@ -426,10 +426,19 @@ func (p *Planner) addAdoptOpenEBSLabels(component *unstructured.Unstructured) er
 // components and their configuration.
 func (p *Planner) getDesiredOpenEBS() error {
 	openebs := &unstructured.Unstructured{}
+	openebsLabels := make(map[string]string)
+	// get the adoption job ID label which tells the ID of the job which has triggered
+	// the adoption process.
+	if p.ObservedAdoptOpenEBS.GetLabels() != nil {
+		if jobID, exist := p.ObservedAdoptOpenEBS.GetLabels()[types.KeyAdoptionJobID]; exist {
+			openebsLabels[types.KeyAdoptionJobID] = jobID
+		}
+	}
 	openebs.SetUnstructuredContent(map[string]interface{}{
 		"metadata": map[string]interface{}{
-			"name":      "my-openebs-123",
+			"name":      "my-openebs-config",
```

Comment:
Should this have uniqueness in the name?

Reply:
I haven't kept it unique or something since this is a one time process and also it can't generate a different name in each reconciliation otherwise it would not be able to identify the CR created by itself.

Reply:
Will it be a problem if the same OpenEBS is adopted twice? Install OpenEBS, connect to director, adopt, disconnect, connect again, adopt.

Reply:
Right @avishnu, what will happen is that it will not create a new OpenEBS CR since we are not allowing update functionalities as of today to the adoption controller since it is a one time job.

Reply:
Why we don't want to have a unique name is that a particular namespace ideally should not contain more than one adoptOpenEBS as well as OpenEBS, if we have unique name criteria then a number of them could get created.

Reply:
Are we going to remove it later? Since this is a one time activity?

Reply:
I am somehow not able to understand from the name "my-openebs-config"
Can this be "adopted-openebs" instead?

Reply:
Will start having a doc.go for code workflow and all.
However, this code is trying to solve the adoption use case, here it is creating an OpenEBS CR with name `my-openebs-config` which means that this CR contains the user's OpenEBS configuration which will be used going further to manage user's OpenEBS.
When I said one-time activity, I meant that OpenEBS adoption itself is a one-time activity hence OpenEBS CR creation by adopt controller also.

I think it is quite clear that an OpenEBS CR represents the OpenEBS configuration of a user and it is namespace scoped since it is namespace scoped, it is desirable or allowed to have multiple OpenEBS CRs across namespaces (however not practical as of today as we can not have multiple OpenEBS installation within a cluster) but at the same time, it is not ideal to have multiple OpenEBS CRs within the same namespace pointing to the same OpenEBS installation.

The above comment was why we should not have the OpenEBS CR name unique instead of having some fixed name, so my take is that since we do not want to experience a situation where we have a number of OpenEBS CRs within same namespace pointing to the same OpenEBS installation, we are having a fixed name.
So what if the user already have OpenEBS CR with the same name in the same namespace, there are 2 options:
1. Users should ideally not have any openebses or adoptopenebses entries while starting the adoption process.
2. The external tools or processes triggering the adoption process should check if openebs or adoptopenebs entries are already present or not, if present, remove them.